kingpin/actors/aws/s3.py

```diff
@@ -521,6 +580,14 @@ def _generate_lifecycle(self, config):
         for c in config:
             self.log.debug('Generating lifecycle rule from foo: %s' % c)
 
+            # You must supply at least 'prefix' or 'filter' in your
+            # lifecycle config. This is tricky to check in the jsonschema, so
+            # we do it here.
+            if not any(k in c for k in ('prefix', 'filter')):
```

Comment:
You can do this with oneOf... https://stackoverflow.com/questions/24023536/how-do-i-require-one-field-or-another-or-one-of-two-others-but-not-all-of-them